From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Tue, 26 Jan 2016 12:24:44 +0300
Subject: [PATCH] ceph: checking for IS_ERR instead of NULL
Git-commit: 1418bf076d08edd47a610ea3844c6f6012949a51
Patch-mainline: v4.5
References: Fate#318586


ceph_osdc_alloc_request() returns NULL on error, it never returns error
pointers.

Fixes: 5be0389dac66 ('ceph: re-send AIO write request when getting -EOLDSNAP error')
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/file.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/ceph/file.c
+++ b/fs/ceph/file.c
@@ -634,8 +634,8 @@ ceph_sync_direct_write(struct kiocb *ioc
 					    ci->i_truncate_seq,
 					    ci->i_truncate_size,
 					    false);
-		if (IS_ERR(req)) {
-			ret = PTR_ERR(req);
+		if (!req) {
+			ret = -ENOMEM;
 			break;
 		}
 
